### PR TITLE
Add function that check NIC unique names.

### LINKED ISSUE #84: NIC unique names
Network interfaces need to have unique names. This should be reflected in the mock.

### PR DESCRIPTION
**Pull request by mgold1234**
## Please describe the change you are making

This PR fixes #84 and add function that check NIC unique names.
add also a test that check user can't create duplicate nic with same name.

## Are you the owner of the code you are sending in, or do you have permission of the owner?
yes

### CODE REVIEW

File: client_nic_create.go [hunk 1]
```diff
@@ -16,6 +16,9 @@ func (o *oVirtClient) CreateNIC(
 	if err := validateNICCreationParameters(vmid, name); err != nil {
 		return nil, err
 	}
+	if err = checkNICExist(o, vmid, name); err != nil {
```
Comment (janosdebugs) [inline client_nic_create.go:19]: This place in the code is the "live" connection. The oVirt Engine already checks this and we don't need to check it on the client side. Instead, #84 is about the mocking client.

File: client_nic_create_test.go [hunk 2]
```diff
@@ -28,3 +28,35 @@ func TestVMNICCreation(t *testing.T) {
...
+	)
+	assertNICCount(t, vm, 0)
+	nic1 := assertCanCreateNIC(
+		t,
+		helper,
+		vm,
+		nicName,
+		ovirtclient.CreateNICParams())
+	assertNICCount(t, vm, 1)
+	err, _ := vm.CreateNIC(
```
Comment (janosdebugs) [inline client_nic_create_test.go:51]: The `err` variable seems to be a misnomer here, the CreateNIC returns a NIC. Also, this section should be moved into a function called `assertCannotCreateNIC` for reusability.

I would also suggest checking the err instead of the returned NIC.

**Review by janosdebugs — CHANGES_REQUESTED** [review]
Hey @mgold1234 thank you very much for your contribution. #84 wasn't explicit enough, that issue is about the mock facility, not the live client. The mock client does not check the names currently, but the live oVirt engine does.

The code in question should be added to the mock client.

**Comment by janosdebugs** [comment]
Lint failures are not related to this PR, overriding and merging.

**Comment by janosdebugs** [comment]
@mgold1234 thank you for your contribution, the PR is now merged and will be in the next go-ovirt-client release.
